Tasks/AzureRmWebAppDeployment/operations/ReleaseAnnotationUtility.ts

```diff
@@ -0,0 +1,76 @@
+import tl = require('vsts-task-lib/task');
+import { AzureAppService } from 'azure-arm-rest/azure-arm-app-service';
+import { AzureApplicationInsights, ApplicationInsightsResources} from 'azure-arm-rest/azure-arm-appinsights';
+import { AzureEndpoint } from 'azure-arm-rest/azureModels';
+var uuidV4 = require("uuid/v4");
+export async function addReleaseAnnotation(endpoint: AzureEndpoint, azureAppService: AzureAppService, isDeploymentSuccess: boolean): Promise<void> {
+    try {
+        var appSettings = await azureAppService.getApplicationSettings();
+        var instrumentationKey = appSettings && appSettings.properties && appSettings.properties.APPINSIGHTS_INSTRUMENTATIONKEY;
+        if(instrumentationKey) {
+            let appinsightsResources: ApplicationInsightsResources = new ApplicationInsightsResources(endpoint);
+            var appInsightsResources = await appinsightsResources.list(null, [`$filter=InstrumentationKey eq '${instrumentationKey}'`]);
+            if(appInsightsResources.length > 0) {
+                var appInsights: AzureApplicationInsights = new AzureApplicationInsights(endpoint, appInsightsResources[0].id.split('/')[4], appInsightsResources[0].name);
```

Comment:
You have already defined a method to get RG, can we use that instead of this logic -  appInsightsResources[0].id.split('/')[4]

Reply:
This RG filtering is specific to App insights with its own API URL. so, not suitable with generic RG filtering.